src/app/components/authorized/post-details/post-details.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { GatewayService } from 'src/app/services/gateway.service';
+import { PostDetails } from 'src/types/posts';
+
+@Component({
+    selector: 'app-post-details',
+    templateUrl: './post-details.component.html',
+    styleUrls: ['./post-details.component.scss'],
+})
+export class PostDetailsComponent implements OnInit {
+    @Input() post: PostDetails;
+
+    constructor(private _route: ActivatedRoute, private _gatewayService: GatewayService) {}
+
+    ngOnInit(): void {
+        this.getPostDetails();
```

Comment:
This naming is not really correct. You are not `get`ting anything. Rather, you are loading data and setting the value. The `get` indicates that there is a value returned, but the function is void.
```
const details = getPostDetails();
// or
setPostDetails();
```
In this case, it may be best to do something like
```
ngOnInit(): void {
    const id = ...;
    this._gatewayService.loadPostDetails(id).subscribe(...);
}
```

I don't think there's good enough reason to have it separate. If you feel there is, I'd suggest at least better naming.

Reply:
Refactored!